src/server/pps/iface.go

```diff
@@ -18,6 +18,6 @@ type APIServer interface {
 	StopJobInTransaction(*txncontext.TransactionContext, *pps_client.StopJobRequest) error
 	UpdateJobStateInTransaction(*txncontext.TransactionContext, *pps_client.UpdateJobStateRequest) error
 	CreatePipelineInTransaction(*txncontext.TransactionContext, *pps_client.CreatePipelineRequest) error
-	InspectPipelineInTransaction(*txncontext.TransactionContext, string) (*pps_client.PipelineInfo, error)
+	InspectPipelineInTransaction(txn *txncontext.TransactionContext, projectName, pipelineAncestry string) (*pps_client.PipelineInfo, error)
```

Comment:
Could we just use a pipeline proto here?

Reply:
I could, but that makes it unclear in to implementers of the interface if the string `Name` member should be treated like a name or a name-with-ancestry-syntax.

Also, by explicitly changing the signature to include the project name I force an update of all callers and implementations to ensure that the project name is taken into account. Otherwise it would be too easy to have a problem like these:

```go
a.InspectPipelineInTransaction(txn, &Pipeline{Name: pipelineName}) // forgot to include the project!
```

or:

```go
func (a someServer) InspectPipelineInTransaction(txn *txncontext.TransactionContext, p *pps.Pipeline) (*pps.PipelineInfo, error) {
    dbQuery(p.Name) // forgot to include the project!
}
```

I’m trying to use the type system to reduce human error.

Reply:
What I said about the ancestry syntax above applies here. Also, I don't agree with using more parameters to potentially reduce bugs. It is less maintainable than a proto, forces clients to set fields that may not matter for the function (that is why config structs and functional options exist), and doesn't do much in terms of preventing bugs because it is still easy to just set the wrong value in the argument. For this particular case, we should ensure that a pipeline proto is created with all the fields correctly set when using it internally.

Reply:
> forces clients to set fields that may not matter for the function

In this case that none of the arguments is optional: a pipeline must never be referred to by pipeline name alone, but always by project name and pipeline name. Passing a Pipeline message permits an implicit missing Project message or an implicit empty Project name.

I changed the signature so that the interface takes a Pipeline message.